src/agent/debuglet.ts

```diff
@@ -322,6 +329,18 @@ export class Debuglet extends EventEmitter {
 
     });
   }
+  checkReady() {
+    const diff = process.hrtime(this.promiseResolvedTimestamp_);
+    if (diff[0] < PROMISE_RESOLVE_CUT_OFF_IN_SECONDS) {
+      return Promise.resolve();
+    }
+    this.promise_ = new Promise<void>((resolve) => {
```

Comment:
I understand that GCF is restricted to calling `checkReady` a second time only after the Promise returned by the first call has been resolved... buthis is not a general restriction -- users can still call `checkReady` arbitrarily. Given this, `this.promiseResolve_` may have its value clobbered. (Also see: https://github.com/GoogleCloudPlatform/cloud-debug-nodejs/pull/341#discussion_r145486623)

Reply:
Good question :)
Per our discussion earlier today, we will return the same promise if the cached promise existed. Otherwise, we will calculate the time since last listBreakpoint and determine if resolve a promise immediately or create a new promise and cache it.